src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -0,0 +1,757 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2017 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import net.sf.cglib.proxy.Enhancer;
+import net.sf.cglib.proxy.MethodInterceptor;
+import net.sf.cglib.proxy.MethodProxy;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.assertj.core.util.CheckReturnValue;
+
+import java.io.File;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicMarkableReference;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.concurrent.atomic.AtomicStampedReference;
+
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+/**
+ * Entry point for assumption methods for different types, which allow to skip test execution on failed assumptions.
+ * @since 2.9.0
+ */
+public class Assumptions {
+
```

Comment:
If we add this test (I think we should for API consistency) it means the we need to add the assertions you mentioned as assumptions:
- `assertThat(Iterable<? extends ELEMENT>, AssertFactory<ELEMENT,ELEMENT_ASSERT>) `
- `assertThat(List<? extends ELEMENT>, AssertFactory<ELEMENT,ELEMENT_ASSERT>) `
- `assertThat(List<? extends ELEMENT>, Class<ELEMENT_ASSERT>) `
- `assertThat(T)  // <T extends Comparable<? super T>> `
- `assertThat(T)  // <T extends AssertDelegateTarget> `
- `assertThat(ACTUAL, java.lang.Class<ELEMENT_ASSERT>) `
- `assertThat(AssertProvider<T>)`